opslevel/resource_opslevel_check_package_version.go

```diff
@@ -0,0 +1,314 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/relvacode/iso8601"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &CheckPackageVersionResource{}
+	_ resource.ResourceWithImportState = &CheckPackageVersionResource{}
+)
+
+func NewCheckPackageVersionResource() resource.Resource {
+	return &CheckPackageVersionResource{}
+}
+
+// CheckPackageVersionResource defines the resource implementation.
+type CheckPackageVersionResource struct {
+	CommonResourceClient
+}
+
+type CheckPackageVersionResourceModel struct {
+	Category    types.String `tfsdk:"category"`
+	Description types.String `tfsdk:"description"`
+	Enabled     types.Bool   `tfsdk:"enabled"`
+	EnableOn    types.String `tfsdk:"enable_on"`
+	Filter      types.String `tfsdk:"filter"`
+	Id          types.String `tfsdk:"id"`
+	Level       types.String `tfsdk:"level"`
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+
+	// TODO: Unique Fields Here
+	MissingPackageResult       types.String `tfsdk:"missing_package_result"`
+	PackageConstraint          types.String `tfsdk:"package_constraint"`
+	PackageManager             types.String `tfsdk:"package_manager"`
+	PackageName                types.String `tfsdk:"package_name"`
+	PackageNameIsRegex         types.Bool   `tfsdk:"package_name_is_regex"`
+	VersionConstraintPredicate types.Object `tfsdk:"version_constraint_predicate"`
+}
+
+func ParsePredicate(predicate *opslevel.Predicate) types.Object {
```

Comment:
This can probably be moved to a utility area and all the other predicates could use it